src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,36 +23,44 @@
             + "Parameters: INDEX (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_PASSENGER_SUCCESS = "Deleted Passenger: %1$s";
+    public static final String MESSAGE_DELETE_PASSENGER_SUCCESS = "Deleted %o Passengers";
     public static final String MESSAGE_DELETE_PASSENGER_FAIL_HAS_POOL = "Failed to delete. One or more Pools contain"
             + " Passenger: %1$s.";
 
-    private final Index targetIndex;
+    private final List<Index> targetIndexes;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(List<Index> targetIndexes) {
+        this.targetIndexes = targetIndexes;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Passenger> lastShownList = model.getFilteredPassengerList();
+        List<Passenger> targetedPassengers = new ArrayList<>();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PASSENGER_DISPLAYED_INDEX);
+        for (Index targetIndex : targetIndexes) {
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PASSENGER_DISPLAYED_INDEX);
+            }
+
+            Passenger passengerToDelete = lastShownList.get(targetIndex.getZeroBased());
+            targetedPassengers.add(passengerToDelete);
         }
 
-        Passenger passengerToDelete = lastShownList.get(targetIndex.getZeroBased());
-        if (!model.deletePassenger(passengerToDelete)) {
-            throw new CommandException(String.format(MESSAGE_DELETE_PASSENGER_FAIL_HAS_POOL, passengerToDelete));
+        for (Passenger p : targetedPassengers) {
+            if (!model.deletePassenger(p)) {
+                throw new CommandException(String.format(MESSAGE_DELETE_PASSENGER_FAIL_HAS_POOL, p));
+            }
         }
-        return new CommandResult(String.format(MESSAGE_DELETE_PASSENGER_SUCCESS, passengerToDelete));
+
+        return new CommandResult(String.format(MESSAGE_DELETE_PASSENGER_SUCCESS, targetedPassengers.size()));
```

Comment:
Should we implement it like this (i.e. delete the passengers at index until we hit one with a pool, ignore the rest)? Perhaps a better implementation is to delete all that don't have a pool then error at the end. Also, I think it would be more intuitive if we printed the deleted passengers names rather than size.

Reply:
Resolved this by changing logic in implementation as discussed